### PR TITLE
Adding blog post on Kafka Streams extension

### PR DESCRIPTION
**Pull request by gunnarmorling**
@gsmet, @cescoffier, @emmanuelbernard, here's a blog post on the KStreams extension. Feedback welcome! Can we publish it today or tomorrow?

### CODE REVIEW

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+@RegisterForReflection
+public class WeatherStation {
+
+    public int id;
+    public String name;
+}
+----
+
+Then build the application using the `native` profile
+(note this requires GraalVM to be installed on your system):
```
Comment (gsmet) [inline _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]: I would add a link to https://quarkus.io/guides/building-native-image-guide .

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+
+[source, xml]
```
Comment (gsmet) [inline _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]: xml?

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+(RSS, resident set size).
+Note that this number also contains the memory needed for the HTTP endpoint defined in that example,
+which is used for answering interactive queries via REST.
+
+== Gaining Operational Insight
+
+One of the nice things about Quarkus is that it comes with support for all the Eclipse MicroProfile APIs.
+They help to address common requirements when building microservices,
+such as https://microprofile.io/project/eclipse/microprofile-health[health checks] ("Is my application running and ready to handle requests?")
+and https://microprofile.io/project/eclipse/microprofile-metrics[metrics] ("What's the through-put of my application?", "How many requests has it processed?" etc.).
```
Comment (gsmet) [inline _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]: I would have written throughput as one word?

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc [hunk 4]
```diff
@@ -0,0 +1,361 @@
...
+        // ...
+    }
+}
+----
+<1> Process all Kafka Streams metrics, using a unique name to register them
+<2> Some string-typed "metrics" must be excluded
+<3> All metrics whose name ends with "total" or "counter" will be exposed as counter-typed metrics
+<4> All other metrics will be exposed as gauge-typed metrics, i.e. plain numeric values
+
+Once the application is started, the metrics will be exposed under `/health`,
```
Comment (gsmet) [inline _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]: `/metrics` maybe?

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+application:stream_metrics_process_total 2866.0
+# HELP application:stream_metrics_poll_latency_avg The average latency of poll operation.
+# TYPE application:stream_metrics_poll_latency_avg gauge
+application:stream_metrics_poll_latency_avg 83.3135593220339
+# ...
+----
+
+From here it's a matter of minutes to set up Prometheus to scrape this target,
+configure a Prometheus data source in Grafana and configure a dashboard for visualizing the metrics of interest to you.
+E.g. the following show a simple dashboard displaying the poll/process/commit rates and latencies as well as the total number of processed events
```
Comment (gsmet) [inline _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]:
```suggestion
E.g. the following shows a simple dashboard displaying the poll/process/commit rates and latencies as well as the total number of processed events
```

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+in the quickstart examle:
```
Comment (gsmet) [inline _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]:
```suggestion
in the quickstart example:
```

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc (hunk 1 above)
Comment (gunnarmorling) [reply _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]: Good idea.

File: _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc (hunk 4 above)
Comment (gunnarmorling) [reply _posts/2019-06-26-kafka-streams-applications-with-quarkus-and-microprofile.adoc]: :)

**Comment by gunnarmorling** [comment]
Thanks for reviewing, @gsmet. Addressed your remarks.

**Pull request merged**